clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/client/RegionScopedTitusClient.java

```diff
@@ -0,0 +1,202 @@
+    }
+
+    private TitusRestAdapter createTitusRestAdapter(TitusRegion titusRegion) {
+        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor((msg) -> LOGGER.info("[{}] {}", titusRegion.getName(), msg));
+        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+        OkHttpClient okHttpClient = new OkHttpClient.Builder()
+                .connectTimeout(connectTimeoutMillis, TimeUnit.MILLISECONDS)
+                .readTimeout(readTimeoutMillis, TimeUnit.MILLISECONDS)
+                .addNetworkInterceptor(interceptor)
+                .build();
+        return new Retrofit.Builder()
+                .baseUrl(titusRegion.getEndpoint())
+                .client(okHttpClient)
+                .addConverterFactory(JacksonConverterFactory.create(objectMapper))
+                .validateEagerly(true)
+                .build()
+                .create(TitusRestAdapter.class);
+    }
+
+
+    // APIs
+    // ------------------------------------------------------------------------------------------
+
+    @Override
+    public Job getJob(String jobId) {
+        return execute(titusRestAdapter.getJob(jobId));
+    }
+
+    private <T> T execute(Call<T> call) {
+        try {
+            retrofit2.Response<T> response = call.execute();
+            if (response.isSuccess()) {
+                return response.body();
+            }
+            throw new RuntimeException("response failed " + response.code() + " " + response.message());
+        } catch (IOException ioe) {
+            throw new RuntimeException(ioe);
+        }
+
+    }
+
+    @Override
+    public Job findJobByName(String jobName) {
+        List<Job> jobs = execute(titusRestAdapter.getJobsByTag(jobName));
+        if (jobs.isEmpty()) {
+            return null;
+        }
+        Collections.sort(jobs, (j1, j2) -> -1 * j1.getSubmittedAt().compareTo(j2.getSubmittedAt()));
+        return jobs.get(0);
+    }
+
+    @Override
+    public List<Job> findJobsByApplication(String application) {
+        return execute(titusRestAdapter.getJobsByTag(application));
+    }
+
+    @Override
+    public String submitJob(SubmitJobRequest submitJobRequest) {
+        JobDescription jobDescription = submitJobRequest.getJobDescription();
+        if (jobDescription.getType() == null) {
+            jobDescription.setType("service");
+        }
+        if (jobDescription.getUser() == null) {
+            jobDescription.setUser("spinnaker");
+        }
+        jobDescription.setEnvParam("account", titusRegion.getAccount());
+        jobDescription.setEnvParam("region", titusRegion.getName());
+        jobDescription.getTags().add(jobDescription.getName());
+        jobDescription.getTags().add(jobDescription.getAppName());
+        SubmitJobResponse response = execute(titusRestAdapter.submitJob(jobDescription));
+        if (response == null) throw new RuntimeException(String.format("Failed to submit a titus job request for %s", jobDescription));
```

Comment:
seems unlikely (it should blow up in conversion I'd expect) - this check was in the original code and I glossed over it